### PR TITLE
[IMPROVEMENT] Add badge

### PR DESCRIPTION
**Pull request by TheFaR8**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [X] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

add badge in svg (instead of previous one using svgwrite)

### CODE REVIEW

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -622,18 +621,32 @@ def start_ci():
...
-        output = os.path.join(parent_dir, 'static', 'svg', 'status-{platform}.svg'.format(platform=test.platform.value))
-        dwg = svgwrite.Drawing(output, profile='full')
-
+        availableon = os.path.join(parent_dir, 'static', 'svg',
+                                   'status-{platform}.svg'.format(platform=test.platform.value))
+        svglocation = os.path.join(parent_dir, 'static', 'img',
+                                   'status', 'status-{platform}.svg'.format(platform=test.platform.value))
         if status == Status.SUCCESS:
-            dwg.add(dwg.text('Passing', insert=(0, 0.2), fill='green'))
+            svglocation.format(status="SUCCESS", platform=test.platform.value)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This won't work. svglocation has no more variables at this point.

File: mod_ci/controllers.py [hunk 2]
```diff
@@ -622,18 +621,32 @@ def start_ci():
...
+
         elif status == Status.FAILURE:
-            dwg.add(dwg.text('Failing', insert=(0, 0.2), fill='red'))
+            svglocation.format(status="FAILURE", platform=test.platform.value)
+
         else:
-            dwg.add(dwg.text('Unknown', insert=(0, 0.2), fill='gray'))
-        dwg.save()
+            svglocation.format(status="ERROR", platform=test.platform.value)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: If you pass in the status into the image, you can just as well supress all branches.

File: mod_ci/controllers.py [hunk 3]
```diff
@@ -765,6 +778,7 @@ def progress_reporter(test_id, token):
                 if status == TestStatus.canceled:
                     state = Status.ERROR
                     message = 'Tests aborted due to an error; please check'
+                    update_build_badge(state, test)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Why was this added?

File: mod_ci/controllers.py [hunk 4]
```diff
@@ -622,19 +621,33 @@ def start_ci():
...
+
         elif status == Status.FAILURE:
-            dwg.add(dwg.text('Failing', insert=(0, 0.2), fill='red'))
-        else:
-            dwg.add(dwg.text('Unknown', insert=(0, 0.2), fill='gray'))
-        dwg.save()
+            availableon.format(status, platform=test.platform.value)
 
+        else:
+            availableon.format(status, platform=test.platform.value)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: The same still applies for this location.

Besides this, a .format requires all parameters to be filled in at once, so your first format for availableon will throw a KeyError.

You can do this without any if/else structures within the outer if.

File: mod_ci/controllers.py [hunk 5]
```diff
@@ -622,19 +621,25 @@ def start_ci():
 
 
 def update_build_badge(status, test):
+    '''
+    Build status badge for current test to be displayed on sample-platform.ccextractor.org
+
+    :param status: current testing status
+    :type status: str
+    :param test: current commit that is tested
+    :type test: str
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This is incorrect. If `test` was a `str`, then `test.test_type` is not a valid code construct.

File: mod_ci/controllers.py [hunk 6]
```diff
@@ -622,19 +621,25 @@ def start_ci():
...
-        if status == Status.SUCCESS:
-            dwg.add(dwg.text('Passing', insert=(0, 0.2), fill='green'))
-        elif status == Status.FAILURE:
-            dwg.add(dwg.text('Failing', insert=(0, 0.2), fill='red'))
-        else:
-            dwg.add(dwg.text('Unknown', insert=(0, 0.2), fill='gray'))
-        dwg.save()
-
+        availableon = os.path.join(parent_dir, 'static', 'svg',
+                                   '{status}-{platform}.svg'.format(status=Status.value, platform=test.platform.value))
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: `Status.value` is not correct here. You use the class, but what about the `status` string you passed in as argument?

File: mod_ci/controllers.py [hunk 7]
```diff
@@ -622,19 +621,25 @@ def start_ci():
 
 
 def update_build_badge(status, test):
+    '''
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Please use `"""` instead of `'''`, as PEP-257 prescribes.

**Comment by TheFaR8** [comment]
Not so sure about the URL routing part though

**Comment by canihavesomecoffee** [comment]
What are you doing? The previous version was near correct, and you just had a couple of small fixes to do.

This new version makes **no** sense whatsoever.

**Comment by TheFaR8** [comment]
Reverted

File: mod_ci/controllers.py [hunk 8]
```diff
@@ -622,19 +621,27 @@ def start_ci():
...
+
+    :param status: current testing status
+    :type status: str
+    :param test: current commit that is tested
+    :type test: Test
+    :return: Nothing.
+    :rtype: None
+    """
     if test.test_type == TestType.commit:
+        test.status = Status()
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Why are you doing this? This makes still no sense. You pass in a `status` as an argument to this method, and then you're not using it?

Comment (TheFaR8) [reply mod_ci/controllers.py]: change ```test.status``` into ```status``` ?

Comment (TheFaR8) [reply mod_ci/controllers.py]: oooh
i understand now about your comment above

i misread ah

Comment (TheFaR8) [reply mod_ci/controllers.py]: changed

thanks :)

**Comment by canihavesomecoffee** [comment]
Please rebase against master, so that the conflicts are gone.

**Comment by canihavesomecoffee** [comment]
Nvm, I'll do it.
